### ISSUE TITLE
"Parallel tests" search query performs badly

### ISSUE THREAD

**Issue by aaronstillwell** [circleci/circleci-docs#4515]
# Description
@rosieyohannan recently messaged me to say that search is currently directing users using the anchor in the URL. E.g searching 'parallel tests' and clicking the second result, entitled 'Running tests in parallel' links to https://circleci.com/docs/2.0/parallelism-faster-jobs/#running-split-tests, which is un-intuitive. I agree and have found this problematic on our docs for a while.

# Location
Search

# Steps to Reproduce
As above

**Comment by aaronstillwell** [comment]
The problem stems from the following code, which leverages the anchor if it is set on the search record.

https://github.com/circleci/circleci-docs/blob/d4446d7ca7a327920f2ff99b40fb53d576aadfa5/src-js/instantsearch.js#L70-L86

**Comment by teesloane** [comment]
From a first glance at this code it looks like we are getting the nearest anchor and making the results jump to that heading. It sounds like if we remove this code, a search result would then... jump directly to the found text?

**Comment by aaronstillwell** [comment]
@teesloane the anchor is stored in the search record. There is no context there as to what has been searched for. So it could be (and often is) totally irrelevant. Or it could be highly relevant. Luck of the draw.

If we remove it, the search result would link to the page. There would be no jumping. I'll try and dig out the code for where this attribute is populated.

**Comment by teesloane** [comment]
Hmm - I'm not sure where is better or worse - thank you for digging into this. Looking forward to setting up something a bit more intuitive, if possible.

**Comment by aaronstillwell** [comment]
Looking at the Algolia jekyll plugin, there's not much flexibility provided to improve things further.

From memory, the way Algolia's [docsearch](https://docsearch.algolia.com/) works is by parsing multiple search records per page, each record being a "section". Results are de-duplicated via the [distinct parameter](https://www.algolia.com/doc/api-reference/api-parameters/distinct/). 

This would allow the page to "jump" to the relevant section.

**Comment by aaronstillwell** [comment]
Ah, my bad - that's actually how the jekyll plugin works too by default, and that's what the `anchor` attribute refers to. 

First of all, when we search "parallel tests", we don't see a very good first result.

<img width="1244" alt="Screenshot 2020-07-03 at 17 41 13" src="https://user-images.githubusercontent.com/6533711/86486876-6b1a2700-bd54-11ea-9848-d6b5c17264f1.png">

This is actually a separate issue. The reason for the saucelabs doc appearing above the parallel testing page is due to the fact that "tests in parallel", which is the title of our parallel tests doc, is slightly different from the saucelabs doc in that "parallel tests" appears as-is.

Algolia has a setting called [minProximity](https://www.algolia.com/doc/api-reference/api-parameters/minProximity/) which allows us to determine how important the _distance_ between matching words is. 

I recall from experience with blog/site search type use cases, this value is often better off less sensitive, e.g "tests in parallel" shouldn't rank lower than "parallel tests". We currently have it configured to the most sensitive value (1). I can open a secondary issue for this.

**Comment by teesloane** [comment]
Hmm - if we're not going to remove the anchor related content originally posted, maybe we could rename this issue and keep the thread going here?

**Comment by aaronstillwell** [comment]
Back to the original issue though, for some reason the search record representing the "running split tests" section is being returned instead of what we'd probably prefer, which is the first record for that page. I will dig into why and report back.

**Comment by rosieyohannan** [comment]
This particular issue with search appears to be fixed now, so closing. Feel free to reopen though!
